src/components/logout/logout-controller.ts

```diff
@@ -0,0 +1,188 @@
+  }
+
+  if (!queryParams.post_logout_redirect_uri) {
+    logger.info(
+      "No post_logout_redirect_uri, redirecting to default logout URI"
+    );
+    return res.redirect(buildRedirectUri(defaultLogoutUrl, stateParam));
+  }
+
+  if (
+    !config
+      .getPostLogoutRedirectUrls()
+      .includes(queryParams.post_logout_redirect_uri as string)
+  ) {
+    logger.info(
+      "Post logout redirect uri not present in client config: " +
+        queryParams.post_logout_redirect_uri
+    );
+    return res.redirect(
+      buildRedirectUri(defaultLogoutUrl, stateParam, {
+        error_code: "invalid_request",
+        error_description:
+          "client registry does not contain post_logout_redirect_uri",
+      })
+    );
+  }
+
+  if (!isValidUrl(queryParams.post_logout_redirect_uri as string)) {
+    return res.redirect(
+      buildRedirectUri(defaultLogoutUrl, stateParam, {
+        error_code: "invalid_request",
+        error_description: "invalid post logout redirect URI",
+      })
+    );
+  }
+
+  res.redirect(
+    buildRedirectUri(queryParams.post_logout_redirect_uri as string, stateParam)
+  );
+};
+
+const isIdTokenSignatureValid = async (idToken: string): Promise<boolean> => {
+  try {
+    const header = decodeProtectedHeader(idToken);
+    if (header.alg === "RS256") {
+      const rsaKey = await publicJwkWithKidFromPrivateKey(
+        RSA_PRIVATE_TOKEN_SIGNING_KEY,
+        RSA_PRIVATE_TOKEN_SIGNING_KEY_ID
+      );
+      await jwtVerify(idToken, rsaKey);
+      return true;
+    } else {
+      const ecKey = await publicJwkWithKidFromPrivateKey(
+        EC_PRIVATE_TOKEN_SIGNING_KEY,
+        EC_PRIVATE_TOKEN_SIGNING_KEY_ID
+      );
+      await jwtVerify(idToken, ecKey);
+      return true;
+    }
+  } catch (error) {
+    logger.error(
+      "Failed to verify signature of ID token: " + (error as Error).message
+    );
+    return false;
+  }
+};
+
+const parseIdTokenHint = (
+  idTokenHint: string
+): {
+  clientId: string | undefined;
+  clientSessionId: string | undefined;
+  rpPairwiseId: string | undefined;
+} => {
+  const jwtClaimsSet = decodeJwt(idTokenHint);
+
+  return {
+    clientId: Array.isArray(jwtClaimsSet.aud)
+      ? jwtClaimsSet.aud[0]
+      : jwtClaimsSet.aud,
+    clientSessionId: jwtClaimsSet.sid as string | undefined,
+    rpPairwiseId: jwtClaimsSet.sub,
+  };
+};
+
+const isValidUrl = (url: string): boolean => {
+  try {
+    new URL(url);
+    return true;
+  } catch (error) {
+    logger.warn(
+      "Post logout redirect uri is not valid url: " + url,
+      "error: " + (error as Error).message
+    );
+    return false;
+  }
+};
+const buildRedirectUri = (
+  baseurl: string,
+  state: string | undefined,
+  errorOpts?: { error_code: string; error_description: string }
+): string => {
+  const queryParams = { ...errorOpts, ...(state && { state }) };
+  return `${baseurl}?${Object.entries(queryParams)
+    .map(([k, v]) => `${k}=${encodeURIComponent(v)}`)
+    .join("&")}`;
+};
```

Comment:
This is fine, but I wonder if there's not something built into javacript that will do it for us?

Reply:
Yeah it's a bit gnarly, I think actually doing something like 
```
new URLSearchParams(obj).toString() 
```
might do it